util.scad

```diff
@@ -90,3 +90,22 @@ module mirrored(vector_set, index = 0) {
     children();
   }
 }
+
+// A shape that can be used for push in parts
+module pin(height, radius) {
+  inner_height = height * 0.875;
+  cap_height = (height - inner_height) / 2;
+  difference() {
+    union() {
+      up(-inner_height/2)
```

Comment:
two things: one, I think there's a `down(x)` you can use. If not, you should make one. And two, I don't think this should be centered along z. I would imagine, the easiest way to place a pin is if its x/y centered and goes up from z=0. Then you can rotate it, and it will be perfectly aligned with the plane of your choice, and you can then place it at the central point of where you want the pin and it will be centered there in the other two planes regardless of pin size